-1

**Failed Tests**:   ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-71a331/39066/summary.html
**COMMIT**: 1764aee9ca43948a00b6c0a9b451fe4b8ae86f18
**CMSSW**: CMSSW_14_1_X_2024-04-24-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44834/39066/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Local Products Rules ..... done
>> Creating project symlinks
>> Entering Package DataFormats/Math
>> Entering Package TrackingTools/AnalyticalJacobians
>> Compile sequence completed for CMSSW CMSSW_14_1_X_2024-04-24-1100
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_1_X_2024-04-24-1100/tmp/el8_amd64_gcc12/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package DataFormats/Math
Entering library rule at DataFormats/Math
>> Compiling  src/DataFormats/Math/src/ExtVec.cc
</pre>

--- a/TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobianEXT.icc
+++ b/TrackingTools/AnalyticalJacobians/src/AnalyticalCurvilinearJacobianEXT.icc
@@ -49,9 +49,13 @@ void AnalyticalCurvilinearJacobian::computeFullJacobian(const GlobalTrajectoryPa
   double gamma = dot(hn, t2);
   Vec3D an = cross3(hn, t2);
 
+#ifdef __clang__
+  Vec4D t = __builtin_shufflevector(t1, t2, 1, 0, 5, 4);
+#else
   typedef unsigned long long v4sl __attribute__((vector_size(32)));
   v4sl mask{1, 0, 5, 4};
   Vec4D t = __builtin_shuffle(t1, t2, mask);
+#endif
   Vec4D tt = t * t;
 
   double au1 = std::sqrt(tt[0] + tt[1]);

--- a/TrackingTools/AnalyticalJacobians/src/JacobianCurvilinearToLocal.cc
+++ b/TrackingTools/AnalyticalJacobians/src/JacobianCurvilinearToLocal.cc
@@ -52,11 +52,11 @@ void JacobianCurvilinearToLocal::compute(Surface::RotationType const& rot,
   if (cosl < 1.e-30)
     cosl = 1.e-30;
   double cosl1 = 1. / cosl;
-  GlobalVector un(-tn.y() * cosl1, tn.x() * cosl1, 0.);
-  GlobalVector vn(-tn.z() * un.y(), tn.z() * un.x(), cosl);
+  const GlobalVector un(-tn.y() * cosl1, tn.x() * cosl1, 0.);
+  const GlobalVector vn(-tn.z() * un.y(), tn.z() * un.x(), cosl);
 
-  auto u = rot.rotate(un.basicVector());
-  auto v = rot.rotate(vn.basicVector());
+  auto const u = rot.rotate(un.basicVector());
+  auto const v = rot.rotate(vn.basicVector());
 
   int j = 0, k = 1, i = 2;
 

